metadata-ingestion/src/datahub/ingestion/graph/client.py

```diff
@@ -248,9 +248,11 @@ def make_rest_sink(
         with DatahubRestSink(PipelineContext(run_id=run_id), sink_config) as sink:
             yield sink
         if sink.report.failures:
+            logger.error(
+                f"Failed to emit {len(sink.report.failures)} records\n{sink.report.as_string()}"
```

Comment:
I'm wondering if `sink.report.as_string()` might be too verbose. 

The report will be printed out once the pipeline finishes, regardless of success, right? If so, printing it here may be redundant.

Reply:
it is only printed if the sink had any failures

it is somewhat verbose, but it's definitely better than not printing anything